shared/utils/relabel-ids.py

```diff
@@ -187,19 +187,23 @@ def _add_cce_id_refs_to_oval_checks(self, idmappingdict):
                 continue
 
             xccdfcceid = idmappingdict[ovalid]
-            if not ssgcommon.cce_is_valid(xccdfcceid):
-                msg = "The CCE ID {0} is not in valid form.".format(xccdfcceid)
-                raise ssgcommon.SSGError(msg)
-
-            # Then append the <reference source="CCE" ref_id="CCE-ID" /> element right
-            # after <description> element of specific OVAL check
-            ccerefelem = ssgcommon.ElementTree.Element(
-                'reference', ref_id=xccdfcceid, source="CCE")
-            ovaldesc.addnext(ccerefelem)
-            # Sanity check if appending succeeded
-            if ccerefelem.getprevious() is not ovaldesc:
-                msg = "Failed to add CCE ID to {0}.".format(ovalid)
-                raise ssgcommon.SSGError(msg)
+            if ssgcommon.cce_is_valid(xccdfcceid):
+                # Then append the <reference source="CCE" ref_id="CCE-ID" /> element right
+                # after <description> element of specific OVAL check
+                ccerefelem = ssgcommon.ElementTree.Element(
+                    'reference', ref_id=xccdfcceid, source="CCE")
+                metadata = rule.find(".//{%s}metadata" % self.CHECK_NAMESPACE)
+                metadata.append(ccerefelem)
+                # Sanity check if appending succeeded
+                for ref in metadata.findall(".//reference"):
+                    if ref.attrib.get("ref_id") == xccdfcceid:
```

Comment:
you can do this in a pythonic way on just one line:
`cce_ref = ref if ref.attrib.get("ref_id" == xccdfcceid else None`

Reply:
@mpreisler always forget the more pythonic way.... guess it shows. :grin: